src/mtcp/mtcp_restart.c

```diff
@@ -104,11 +104,13 @@ typedef struct RestoreInfo {
   VA vvarStart;
   VA vvarEnd;
   fnptr_t post_restart;
+  fnptr_t post_restart_debug;
```

Comment:
For the Intel compiler, we use inline assembly to compute the offset of `restorememoryareas_fptr` and jump into the function. See the code in `restart_fast_path()`:
```C
      asm volatile(
      // 96 = offsetof(RestoreInfo, rinfo.restorememoryareas_fptr)
      // NOTE: Update the offset when adding fields to the RestoreInfo struct
      "movq    96+rinfo(%%rip), %%rdx;" /* rinfo.restorememoryareas_fptr */
      "leaq    rinfo(%%rip), %%rdi;"    /* &rinfo */
      "movl    $0, %%eax;"
      "call    *%%rdx"
      : : );
```
However, because of the additional field `fnptr_t post_restart_debug;` before `fnptr_t restorememoryareas_fptr;` the offset of `restorememoryareas_fptr` within the struct will change, and so we need to update the code in `restart_fast_path()` to correspond.

Also, see commit: 7074b3a3, where we last fixed this bug for the Intel compiler.